distributed/comm/core.py

```diff
@@ -175,6 +202,26 @@ async def _():
 
         return _().__await__()
 
+    async def on_connection(self, comm: Comm, handshake_overrides={}):
```

Comment:
Nitpick: I'd feel a little better about initializing `handshake_overrides=None` and then assigning it to be `{}` if `handshake_overrides is None` to avoid any issues with using mutable objects as a keyword default

Reply:
In general I agree, but in this case I think that we're safe.

Reply:
I'm interpretting @jcrist 's :+1: as a vote towards safety.  I've gone ahead and changed these to `None`.